Content.Server/GameTicking/GamePresets/PresetTraitorDeathMatch.cs

```diff
@@ -0,0 +1,220 @@
+                    vItem.Owner.Delete();
+
+            // Replace their items:
+
+            //  pda
+            var newPDA = _entityManager.SpawnEntity(PDAPrototypeName, mind.OwnedEntity.Transform.Coordinates);
+            inventory.Equip(EquipmentSlotDefines.Slots.IDCARD, newPDA.GetComponent<ItemComponent>());
+
+            //  belt
+            var newTmp = _entityManager.SpawnEntity(BeltPrototypeName, mind.OwnedEntity.Transform.Coordinates);
+            inventory.Equip(EquipmentSlotDefines.Slots.BELT, newTmp.GetComponent<ItemComponent>());
+
+            //  backpack
+            newTmp = _entityManager.SpawnEntity(BackpackPrototypeName, mind.OwnedEntity.Transform.Coordinates);
+            inventory.Equip(EquipmentSlotDefines.Slots.BACKPACK, newTmp.GetComponent<ItemComponent>());
+
+            // Like normal traitors, they need access to a traitor account.
+            var uplinkAccount = new UplinkAccount(mind.OwnedEntity.Uid, startingBalance);
+            var pdaComponent = newPDA.GetComponent<PDAComponent>();
+            pdaComponent.InitUplinkAccount(uplinkAccount);
+            _allOriginalNames[uplinkAccount] = mind.OwnedEntity.Name;
+
+            // The PDA needs to be marked with the correct owner.
+            pdaComponent.SetPDAOwner(mind.OwnedEntity.Name);
+            newPDA.AddComponent<TraitorDeathMatchReliableOwnerTagComponent>().UserId = mind.UserId;
+
+            // Finally, it would be preferrable if they spawned as far away from other players as reasonably possible.
+            if (FindAnyIsolatedSpawnLocation(mind, out var bestTarget))
+            {
+                mind.OwnedEntity.Transform.Coordinates = bestTarget;
+            }
+            else
+            {
+                // The station is too drained of air to safely continue.
+                if (_safeToEndRound)
+                {
+                    _chatManager.DispatchServerAnnouncement(Loc.GetString("The station is too unsafe to continue. You have one minute."));
+                    _restarter.RoundMaxTime = TimeSpan.FromMinutes(1);
+                    _restarter.RestartTimer();
+                    _safeToEndRound = false;
+                }
+            }
+        }
+
+        // It would be nice if this function were moved to some generic helpers class.
+        private bool FindAnyIsolatedSpawnLocation(Mind ignoreMe, out EntityCoordinates bestTarget)
+        {
+            // Collate people to avoid...
+            var existingPlayerPoints = new List<EntityCoordinates>();
+            foreach (var player in _playerManager.GetAllPlayers())
+            {
+                var avoidMeMind = player.Data.ContentData()?.Mind;
+                if ((avoidMeMind == null) || (avoidMeMind == ignoreMe))
+                    continue;
+                if (avoidMeMind.OwnedEntity == null)
+                    continue;
+                existingPlayerPoints.Add(avoidMeMind.OwnedEntity.Transform.Coordinates);
+            }
```

Comment:
This takes ghosts into account, doesn't it?

Reply:
...right, ghosts. Forgot they existed. They'll end up messing up the spawn locations. Will fix.